pnc_cli/buildconfigurationsets.py

```diff
@@ -301,3 +301,57 @@ def latest_build_set_records_status(id=None, name=None):
     if len(data_json) > 0:
         data_json.sort(key=lambda obj: obj['id'], reverse=True)
         return "Build Config Set Record #" + str(data_json[0]['id']) + ": " + data_json[0]['status']
+
+@arg("-i", "--id", help="ID of the BuildConfigurationSet to build.", type=types.existing_bc_set_id)
+@arg("-n", "--name", help="Name of the BuildConfigurationSet", type=types.existing_bc_set_name)
+@arg("--temporary-build", help="Temporary builds")
+@arg("--timestamp-alignment", help="Enable timestamp alignment for the temporary builds")
+@arg("-f", "--force", help="Force rebuild of all configurations")
+@arg("--id-revisions", nargs="+", help="Specify exact revisions of BuildConfigurations used in set.\n Format: int:int (bcID:revID)", default=None)
+def build_versioned(id=None, name=None, temporary_build=False, timestamp_alignment=False,
+              force=False, **kwargs):
+    """
+    Start a build of the given BuildConfigurationSet with versioned BCs
+    """
+    content = build_versioned_raw(id,name, temporary_build, timestamp_alignment, force,
+                            **kwargs)
+    if content:
+        return utils.format_json(content)
+
+def build_versioned_raw(id=None,name=None, temporary_build=False, timestamp_alignment=False, force=False,
+                        **kwargs):
+    id = common.set_id(pnc_api.build_group_configs, id, name)
+
+    logging.debug("temp_build: " + str(force))
+    logging.debug("timestamp_alignment: " + str(temporary_build))
+    logging.debug("force: " + str(timestamp_alignment))
+
+    if temporary_build is False and timestamp_alignment is True:
+        logging.error("You can only activate timestamp alignment with the temporary build flag!")
+        sys.exit(1)
+
+    revisions = kwargs.get("id_revisions")
+    id_revs = map(parse_revision, revisions) if revisions else None
+
+    bcsRest = common.get_entity(pnc_api.build_group_configs, id)
+    body = swagger_client.BuildConfigurationSetWithAuditedBCsRest()
+    body = fillBCSWithAuditedBody(body, bcsRest, id_revs)
+
+    response = utils.checked_api_call(pnc_api.build_group_configs, 'build_versioned', id=id,
+                                      temporary_build=temporary_build,
+                                      timestamp_alignment=timestamp_alignment,
+                                      force_rebuild=force,
+                                      body=body)
+    if response:
+        return response.content
+
+def parse_revision(revision):
```

Comment:
I wanted to have this somewhere separate, but it is prolly the only case where this func and the one at :352 line will get used so i wanted to keep them here. I also thought of getting them private ( "__" thingy before the name).